### PR TITLE
Change CircleCI to GitHub Actions

### PR DESCRIPTION
**Pull request by koji-1009**
I suggest updates CI script and using GitHub Actions.
The result looks like [this](https://github.com/koji-1009/groupie/actions/runs/491880471).
ref #335

---

Currently, unit test and lint are not checked when creating a PR.
Therefore, I changed it so that it is executed every PR.

---

If you want to display Android Lint's suggestions in PR, it's a good idea to set up Danger.
With GitHub Actions, it's easy to add Danger's suggestions, so I think it's an advantage to move to GitHub Actions.

Thanks!

### CODE REVIEW

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Unit tests
+        run: ./gradlew test --stacktrace
+      - uses: actions/upload-artifact@v2
+        with:
+          name: unit tests (Java ${{ matrix.java }})
+          path: library/build/reports/tests
```
Comment (koji-1009) [inline .github/workflows/pull_request.yml:35]: ref <https://developer.android.com/studio/test/command-line>

**Comment by Zhuinden** [comment]
@msya can you by chance check if this is correct and if it retains behavior we currently have?

**Comment by koji-1009** [comment]
If the review is difficult, I will revert the CircleCI's fix. 
That way, we just discuss whether to add a workflow.

**Comment by koji-1009** [comment]
As the need for this patch seems low, I will close it. Thanks.

**Pull request closed without merging**